src/metrics/metric_types.go

```diff
@@ -20,3 +24,61 @@ func (qd QueryDefinition) GetDataModels() interface{} {
 	ptr := reflect.New(reflect.ValueOf(qd.dataModels).Type())
 	return ptr.Interface()
 }
+
+func (qd *QueryDefinition) insertDatabaseNames(databases collection.DatabaseList) *QueryDefinition {
+	databaseList := ""
+	for database := range databases {
+		databaseList += `'` + database + `',`
+	}
+	databaseList = databaseList[0 : len(databaseList)-1]
+
+	qd.query = strings.Replace(qd.query, `%DATABASES%`, databaseList, 1)
+
+	return qd
+}
+
+func (qd *QueryDefinition) insertSchemaTables(schemaList collection.SchemaList) *QueryDefinition {
+	schemaTables := make([]string, 0)
+	for schema, tableList := range schemaList {
+		for table := range tableList {
+			schemaTables = append(schemaTables, fmt.Sprintf("'%s.%s'", schema, table))
+		}
+	}
+
+	if len(schemaTables) == 0 {
+		return nil
+	}
+
+	schemaTablesString := strings.Join(schemaTables, ",")
+
+	newTableDef := &QueryDefinition{
+		dataModels: qd.dataModels,
+		query:      strings.Replace(qd.query, `%SCHEMA_TABLES%`, schemaTablesString, 1),
+	}
+
+	return newTableDef
+}
+
+func (qd *QueryDefinition) insertSchemaTableIndexes(schemaList collection.SchemaList) *QueryDefinition {
```

Comment:
I have some troubles understanding if the code was simply moved or also modified, would make sense to split those two actions in different commits?

Reply:
Absolutely right. Should have kept it in different commits. They were moved and modified :(